src/lib/coapcore/src/error.rs

```diff
@@ -12,7 +12,7 @@ pub struct CredentialError {
 
 #[derive(Debug, Copy, Clone)]
 #[non_exhaustive]
```

Comment:
```suggestion
/// Classification of a [`CredentialError`].
///
/// The variants in here are mainly used for debug output, and all signify *that*
/// the processing of the token was not successful. This type can be used to
/// construct a [`CredentialError`].
#[derive(Debug, Copy, Clone)]
#[non_exhaustive]
```

CI complained about missing documentation; let's see if there's more on the next run.